src/tactic/core.lean

```diff
@@ -1612,6 +1612,43 @@ add_tactic_doc
 
 attribute [higher_order map_comp_pure] map_pure
 
+/-- Attribute `needed` is used to mark declarations that are featured
+in the current file.  Then, the `#unneeded` command can be used to
+list the declaration present in the file that do not support the main
+features of the file. -/
+@[user_attribute]
+meta def needed_attr : user_attribute :=
+{ name := `needed,
+  descr := "tag essential declarations to help identify unused definitions" }
+
+/-- `unneeded n m` removes from the map of unneeded declarations those
+referenced by declaration named `n` which is considerred to be a
+needed declaration -/
+private meta def unneeded : name → name_map declaration → tactic (name_map declaration)
```

Comment:
I'm confused by the naming here. This function is updating the list of unneeded declarations when something is specified as needed, right? A name that makes this "updating" nature clear would help.